pkg/crc/config/viper_config.go

```diff
@@ -34,6 +38,42 @@ func NewViperStorage(configFile, envPrefix string) (*ViperStorage, error) {
 		return nil, fmt.Errorf("error reading configuration file '%s': %v", configFile, err)
 	}
 	v.WatchConfig()
+	// only watch config file if not running in a terminal/interactively
+	if !terminal.IsTerminal(int(os.Stdout.Fd())) {
+		go func() {
+			var tempConfig map[string]interface{} = make(map[string]interface{})
+			w, err := fsnotify.NewWatcher()
+			if err != nil {
```

Comment:
Have you considered using `func (v *Viper) OnConfigChange(run func(in fsnotify.Event))` instead of reimplementing the whole file watching?

Reply:
Yes, i did try using `OnConfigChange` at first, but this only gets called if we have the `viper.WatchConfig`, and `viper.WatchConfig` automatically calls `ReadInConfig`, which is supposed to update the config but it doesn't.

So i avoided using these two methods and directly reimplement it in here.

Reply:
Even if `viper.WatchConfig` is not working as expected, wouldn't it be less code to use `WatchConfig` + `OnConfigChange` rather than having a watch ourselves?

Reply:
Yes, it'll be less code, will switch to using `WatchConfig` + `OnConfigChange`